### PR TITLE
feat: expose `max_batching_rows` in `remote_function`

### PR DESCRIPTION
**Pull request by shobsi**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes internal issue 335506062 🦕

### CODE REVIEW

File: bigframes/functions/remote_function.py [hunk 1]
```diff
@@ -723,6 +747,15 @@ def remote_function(
             projects/PROJECT_ID/locations/LOCATION/repositories/REPOSITORY_NAME.
             For more details see
             https://cloud.google.com/functions/docs/securing/cmek#before_you_begin.
+        max_batching_rows (str, Optional):
```
Comment (GarrettWu) [inline bigframes/functions/remote_function.py:750]: should it be int?

File: bigframes/session/__init__.py [hunk 2]
```diff
@@ -1635,6 +1636,15 @@ def remote_function(
                 projects/PROJECT_ID/locations/LOCATION/repositories/REPOSITORY_NAME.
                 For more details see
                 https://cloud.google.com/functions/docs/securing/cmek#before_you_begin.
+            max_batching_rows (str, Optional):
```
Comment (GarrettWu) [inline bigframes/session/__init__.py:1639]: int

File: tests/system/large/test_remote_function.py [hunk 3]
```diff
@@ -1300,3 +1300,34 @@ def square_num(x):
...
+
+@pytest.mark.parametrize(
+    ("max_batching_rows"),
+    [
+        10_000,
+        None,
+    ],
+)
+@pytest.mark.flaky(retries=2, delay=120)
+def test_remote_function_max_batching_rows(session, scalars_dfs, max_batching_rows):
```
Comment (GarrettWu) [inline tests/system/large/test_remote_function.py:1313]: Would be good if we can verify the config of the created function.

**Review by GarrettWu — APPROVED** [review]

File: bigframes/functions/remote_function.py (hunk 1 above)
Comment (shobsi) [reply bigframes/functions/remote_function.py]: Thanks for catching, corrected

File: bigframes/session/__init__.py (hunk 2 above)
Comment (shobsi) [reply bigframes/session/__init__.py]: corrected

File: tests/system/large/test_remote_function.py (hunk 3 above)
Comment (shobsi) [reply tests/system/large/test_remote_function.py:1313]: makes sens, done, thanks.

File: tests/system/large/test_remote_function.py [hunk 4]
```diff
@@ -1320,6 +1320,11 @@ def square(x):
             [int], int, reuse=False, max_batching_rows=max_batching_rows
         )(square)
 
+        bq_routine = session.bqclient.get_routine(
+            square_remote.bigframes_remote_function
+        )
+        bq_routine.remote_function_options.max_batching_rows == max_batching_rows
```
Comment (GarrettWu) [inline tests/system/large/test_remote_function.py:1326]: assert?

Comment (shobsi) [reply tests/system/large/test_remote_function.py]: corrected, thanks for the catch

**Pull request merged**